CONTRIBUTING.md

```diff
@@ -109,25 +109,109 @@ The workflow is documented in greater detail in the [repository management](./do
 
-## Releasing packages
+## Managing packages
+
+This repository uses [lerna](https://lernajs.io) to manage Gutenberg modules and publish them as packages to `npm`. 
+
+### Creating new package
+
+When creating a new package you need to provide at least the following:
+
+1. `package.json` based on the template:
+	```json
+	{
+		"name": "@wordpress/package-name",
+		"version": "1.0.0-beta.0",
+		"description": "Package description.",
+		"author": "The WordPress Contributors",
+		"license": "GPL-2.0-or-later",
+		"keywords": [
+			"wordpress"
+		],
+		"homepage": "https://github.com/WordPress/gutenberg/tree/master/packages/package-name/README.md",
+		"repository": {
+			"type": "git",
+			"url": "https://github.com/WordPress/gutenberg.git"
+		},
+		"bugs": {
+			"url": "https://github.com/WordPress/gutenberg/issues"
+		},
+		"main": "build/index.js",
+		"module": "build-module/index.js",
+		"react-native": "src/index",
+		"dependencies": {
+			"@babel/runtime-corejs2": "7.0.0-beta.56"
+		},
+		"publishConfig": {
+			"access": "public"
+		}
+	}
+	```
+	It assumes that code is goind to be located in `src` folder and will be transpiled with `Babel`.
+2. `.npmrc` file which disables creating `package-lock.json` file for the package:
+	```
+	package-lock=false
+	```
+3. `README.md` file containing at least:
+	- Package name.
+	- Package description.
+	- Installation details.
+	- Usage example.
+	- `Code is Poetry` logo.
+	
+### Maintaining changelogs
+
+It isn't easy task to maintain dozens of npm packages. That's why we decided to introduce `CHANGELOG.md` files for all packages to simplify the release process.
+
+The developer who proposes a change (pull request) is responsible to choose the correct version increment (`major`, `minor` or `patch`) according to the following guidelines:
+- Major version X (X.y.z | X > 0) should be bumped on any backwards-incompatible change. This will usually occur at the final stage of deprecating and the removal of a feature. The deprecation should be fully backwards-compatible and, if it is not, it should warrant a separate major version bump.
+- Minor version Y (x.Y.z | x > 0) update should be selected when you add functionality in a backwards-compatible manner. It must be incremented if any public API functionality is marked as deprecated.
+- Patch version Z (x.y.Z | x > 0) should be incremented when you make backwards-compatible bug fixes.
+
+When in doubt, refer to [Semantic Versioning specification](https://semver.org/).
+
+_Example:_
+
+```md
+## v1.2.2 (Unreleased)
+
+#### Bug Fix
+
+- ...
+- ...
+```
+
+- If you need to add something considered a bug fix, you add the item to `Bug Fix`section and leave the version as 1.2.2.
+- If it's a new feature you add the item to `New Feature` section and change version to 1.3.0.
+- If it's a breaking change you want to introduce, you add the item to `Breaking Change` section and bump the version to 2.0.0.
```

Comment:
Are there any other changes which wouldn't fall neatly under one of these classifications? I've personally observed some "Internal" / "Housekeeping" type changes which don't necessarily contribute to public-facing consumer, though I suppose could have some place in the SemVer hierarchy. These are also items it's not entirely clear we'd need to include at all.

Reply:
> These are also items it's not entirely clear we'd need to include at all. 

It might get tricky because Lerna will bump version anyways. We probably should skip them as they aren't that important but we still will have a new version released. It might sometimes happen only because one of the packages maintained by Lerna listed in dependencies has version bump.